helm-chart/flink-operator/update_template.sh

```diff
@@ -1,17 +1,84 @@
-sed -i '/- \.\.\/crd/d' ../../config/default/kustomization.yaml
-kubectl kustomize ../../config/default | tee templates/flink-operator.yaml
-sed -i '1s/^/{{- if .Values.rbac.create }}\n/' templates/flink-operator.yaml
-sed -i -e "\$a{{- end }}\n" templates/flink-operator.yaml
-sed -i 's/'"${NS}"'/{{ .Values.flinkOperatorNamespace.name }}/g' templates/flink-operator.yaml
-sed -i 's/replicas: 1/replicas: {{ .Values.replicas }}/g' templates/flink-operator.yaml
-sed -i "s/$IMG/{{ .Values.operatorImage.name }}/g" templates/flink-operator.yaml
-sed -i 's/--watch-namespace=/--watch-namespace={{ .Values.watchNamespace.name }}/' templates/flink-operator.yaml
-cp ../../config/crd/bases/flinkoperator.k8s.io_flinkclusters.yaml ../../config/crd/bases
-
-git checkout ../../config/default/kustomization.yaml
+manifests=$(mktemp /tmp/flink-operator-manifests.yaml)
+
+function yqi() {
+  yq -i "$1" "$manifests"
+}
+
+function modifyManifests() {
+  deploymentSelector='select(.kind == "Deployment")'
+  containersSelector="$deploymentSelector.spec.template.spec.containers"
+  managerSelector="($containersSelector | .[] | select(.name == \"manager\"))"
+  rbacProxySelector="($containersSelector | .[] | select(.name == \"kube-rbac-proxy\"))"
+
+  yqi "$rbacProxySelector"'.image = "__RBAC_PROXY_IMAGE__"'
+  yqi "$rbacProxySelector"'.imagePullPolicy = "__RBAC_PROXY_IMAGE_PULL_POLICY__"'
+  yqi "del($rbacProxySelector.resources)"
+  yqi "$rbacProxySelector |= sort_keys(.)"
+
+  yqi "$managerSelector"'.args += "--watch-namespace=__WATCH_NAMESPACE__"'
+  yqi "$managerSelector"'.resources.limits.cpu = "__LIMITS_CPU__"'
+  yqi "$managerSelector"'.resources.limits.memory = "__LIMITS_MEMORY__"'
+  yqi "$managerSelector"'.resources.requests.cpu = "__REQUESTS_CPU__"'
+  yqi "$managerSelector"'.resources.requests.memory = "__REQUESTS_MEMORY__"'
+  yqi "$managerSelector"'.image = "__MANAGER_IMAGE__"'
+  yqi "$managerSelector"'.imagePullPolicy = "__MANAGER_IMAGE_PULL_POLICY__"'
+  yqi "$managerSelector"' |= sort_keys(.)'
+
+  yqi "$containersSelector |= sort_by(.name)"
+  yqi "$managerSelector"'.name = "flink-operator"'
+  yqi "$deploymentSelector"'.spec.template.metadata.annotations["kubectl.kubernetes.io/default-container"] = "flink-operator"'
+
+  yqi "$deploymentSelector"'.spec.replicas = "__REPLICAS__"'
+  yqi "$deploymentSelector"'.spec.template.spec.serviceAccountName = "__SERVICE_ACCOUNT__"'
+  yqi '(select(.kind == "ClusterRoleBinding" or .kind == "RoleBinding").subjects[] | select(.kind == "ServiceAccount")).name = "__SERVICE_ACCOUNT__"'
+  yqi '(select(.kind == "ClusterRoleBinding" or .kind == "RoleBinding").subjects[] | select(.kind == "ServiceAccount")).namespace = "__NAMESPACE__"'
+  yqi 'select(.metadata.namespace != null).metadata.namespace = "__NAMESPACE__"'
+  yqi 'select(.kind == "CustomResourceDefinition").spec.conversion.webhook.clientConfig.service.namespace = "__NAMESPACE__"'
+  yqi 'del(.metadata.annotations["cert-manager.io/inject-ca-from"])'
```

Comment:
Hi @nickcaballero Thanks for tackling this. So here he are deleting the cert-manager annotations but using cert-manager is what we want. We want to move away from self signed certs.

Reply:
Makes sense. I can remove them now I suppose, or should it wait until cert-manager is part of the chart?

Reply:
I think the main idea would be to have cert-manager part of the chart. I guess it could be in a separate PR.

Reply:
Do you think you can give it a try? 🙏

Reply:
Alright, I've replaced the old `generate-cert.yaml` with a `webhook.yaml` sourced from the Kustomize output.